In [100]:
# import os

ファイルを読み込み、input_text[]に１行ずつ格納する。

In [101]:
input_text = {}

with  open('largev3_平屋建築.txt', encoding="utf-8") as input_file:
    for i, line in enumerate(input_file):
        #print(i," ",line)
        input_text[i] = line

for i in input_text:
    print(i," ", input_text[i])

print(len(input_text))



0   [0.00s -> 4.26s] これを張ってない会社っていうのはまずないはずなんだよね

1   [4.26s -> 7.66s] やっぱり現場確認するっていうのはとても重要になってきます

2   [7.66s -> 9.52s] ここは全部換気扇だと思ってください

3   [9.52s -> 14.28s] 家作りや人生作り

4   [14.28s -> 16.10s] 皆さんどうも職人社長の平松です

5   [16.10s -> 18.24s] 平松社長また現場に来てますね

6   [18.24s -> 18.60s] そうですね

7   [18.60s -> 21.36s] 今回ですね上等からしばらく経って

8   [21.36s -> 25.68s] 断熱とか施工とかですね防水関係とかっていうところの

9   [25.68s -> 28.14s] いろんな重要なポイントを解説プラス

10   [28.14s -> 30.90s] 実際手を動かして施工していきたいと思います

11   [30.90s -> 33.22s] その手前の段階でシリーズになってるんで

12   [33.22s -> 36.04s] その前の内容を知りたいというか見たいという方はですね

13   [36.04s -> 38.62s] 概要欄の方に貼っておくのでそちらのまた見ていくと

14   [38.62s -> 40.70s] こういう感じで済んでるんだなって分かると思うので

15   [40.70s -> 42.72s] またそちらもぜひ見ていただければと思います

16   [42.72s -> 44.70s] ということで早速ですね今回は

17   [44.70s -> 49.18s] 断熱の施工の仕方とか通気の取り方とか

18   [49.18s -> 53.88s] 断熱性耐久性省エネ性能とかに大きく影響があるようなですね

19   [53.88s -> 56.70s] 細かい施工ちょっとの違いで大きく差が出てくるんですよ

20   [56.70s -> 57.86s] っていうところが分かるようにね

21   [57.86s -> 60.14s] 解説をして手を動かしてやっていきますので

22   [60.14s -> 6

## 時間情報入りテキストの処理用 ここから
テキストを時間情報と文章に分離する。

In [102]:
time = []
input_text2 = []

for i in range(len(input_text)):
    i, text = input_text[i].split("] ")
    print(i[1:],text)
    time.append(i[1:])
    input_text2.append(text)

# print(time, " ", input_text2)

# input_text(入力テキスト（時間情報入り）)をinput_text2（時間情報なし）の内容に入れ替え
input_text = input_text2

input_text[0]

0.00s -> 4.26s これを張ってない会社っていうのはまずないはずなんだよね

4.26s -> 7.66s やっぱり現場確認するっていうのはとても重要になってきます

7.66s -> 9.52s ここは全部換気扇だと思ってください

9.52s -> 14.28s 家作りや人生作り

14.28s -> 16.10s 皆さんどうも職人社長の平松です

16.10s -> 18.24s 平松社長また現場に来てますね

18.24s -> 18.60s そうですね

18.60s -> 21.36s 今回ですね上等からしばらく経って

21.36s -> 25.68s 断熱とか施工とかですね防水関係とかっていうところの

25.68s -> 28.14s いろんな重要なポイントを解説プラス

28.14s -> 30.90s 実際手を動かして施工していきたいと思います

30.90s -> 33.22s その手前の段階でシリーズになってるんで

33.22s -> 36.04s その前の内容を知りたいというか見たいという方はですね

36.04s -> 38.62s 概要欄の方に貼っておくのでそちらのまた見ていくと

38.62s -> 40.70s こういう感じで済んでるんだなって分かると思うので

40.70s -> 42.72s またそちらもぜひ見ていただければと思います

42.72s -> 44.70s ということで早速ですね今回は

44.70s -> 49.18s 断熱の施工の仕方とか通気の取り方とか

49.18s -> 53.88s 断熱性耐久性省エネ性能とかに大きく影響があるようなですね

53.88s -> 56.70s 細かい施工ちょっとの違いで大きく差が出てくるんですよ

56.70s -> 57.86s っていうところが分かるようにね

57.86s -> 60.14s 解説をして手を動かしてやっていきますので

60.14s -> 61.62s 早速スタートしていきましょう

61.62s -> 63.50s 早速見ていきましょうの前に

63.50s -> 66.74s 12月28日に門川さんからスマイタイゼンという本が

66.74s -> 68.16s 出版されることになりました

68.16s -> 69.30s おめで

'これを張ってない会社っていうのはまずないはずなんだよね\n'

In [103]:
input_text[100]

'この上に壁を張ったりすると\n'

## 時間情報入りテキストの処理用 ここまで

格納したデータの各行にキーワードがあるかどうか確認し、ある行の番号をkeyword_line[]に格納する。


In [104]:
search_text = "施工" ############ 検索する言葉 ###########


keyword_line = []

for i in range(len(input_text)): #### ←修正した
    this_line = input_text[i]
    if search_text in this_line:
        print(i, " ", this_line)
        keyword_line.append(i)

print(keyword_line)

8   断熱とか施工とかですね防水関係とかっていうところの

10   実際手を動かして施工していきたいと思います

17   断熱の施工の仕方とか通気の取り方とか

19   細かい施工ちょっとの違いで大きく差が出てくるんですよ

199   施工は逆に下から上が原則なんですよ

216   施工は全部下から上っていうのは原則です

257   あと施工する電気屋さんとか

259   この後の施工大事なんで

271   当然先と同じように施工するってこと

280   工事の施工っていうのは

282   施工制度よくやるんだけど

340   丁寧に施工されてます

351   施工で直してくださいって

430   その施工がしっかりしてたら多分大丈夫だよねっていう

435   そうですね施工の方法とか今回ポイントいろいろ伝えてるんで

572   そうそう土台の施工の動画また見てもらえれば

634   よっぽど施工がひどすぎない限りは起こらない

636   とんでもなく雑な施工した時に起こるっていうのは

638   このように丁寧に施工していて

959   ちゃんとこれ施工してますかっていうのを

968   大工さんが施工してチェックして

972   監査員の方が施工で何か不具合あったら

975   それを受けたら施工をもちろん直して写真を撮って

[8, 10, 17, 19, 199, 216, 257, 259, 271, 280, 282, 340, 351, 430, 435, 572, 634, 636, 638, 959, 968, 972, 975]


キーワードがある前後の数十行（zengo_linesで指定）を、決める。

In [105]:
zengo_lines = 10  ##### キーワードがある行の前後何行分かを指定する。

# キーワードがある領域を調べ、keyword_area 配列に０，１で入れる。
# 0: キーワードなしエリア、1:キーワードありエリア


# キーワードがある領域を示す配列の初期化　
keyword_area = [0 for i in enumerate(input_text)]
print(len(keyword_area))
print(len(input_text))

# キーワードがある領域を１にする。
# キーワードがある領域は、キーワードがある行の前後zengo_lines行とする。
for i in range(len(keyword_line)):
    print(i, " ",keyword_line[i])
    for j in range(max(0, keyword_line[i]-zengo_lines),min(len(input_text), keyword_line[i]+zengo_lines)):
        keyword_area[j] = 1

print(keyword_area)



1044
1044
0   8
1   10
2   17
3   19
4   199
5   216
6   257
7   259
8   271
9   280
10   282
11   340
12   351
13   430
14   435
15   572
16   634
17   636
18   638
19   959
20   968
21   972
22   975
[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,

キーワード領域のテキストを結合し、キーワード前後の文脈がわかる文章を作成する。

In [106]:
keyword_sentence = []

print(len(keyword_area))

sentence_end = 0
sentence_start = 1
text_num = 0

text = ""


keyword_sentence_start = []
keyword_sentence_end = []

for i in range(len(keyword_area)):
# for i in range(10):
    if keyword_area[i] == 1: # キーワード領域であるならば
        # text  += input_text[i]
        # print(text)

        if sentence_start == 1:  # キーワード領域の最初の文であるならば
            keyword_sentence.append("...") # 
            sentence_start = 0

            #### 時間の取得
            keyword_sentence_start.append(time[i].split("s -> ")[0])
            print("start_time = ", keyword_sentence_start[text_num])

        if i == len(keyword_area)-1 : ##### 最後のテキストが1である場合の終了時間処理
            keyword_sentence_end.append(time[i].split("s -> ")[1].split("s")[0])
            
        keyword_sentence[text_num] += input_text[i]
    else:
        if sentence_start == 0:
            keyword_sentence[text_num] += "..."

            #### 時間の取得
            keyword_sentence_end.append(time[i].split("s -> ")[1].split("s")[0])
            print("end_time = ", keyword_sentence_end[text_num])



            text_num += 1



        sentence_start = 1 
        # print(".")


1044
start_time =  0.00
end_time =  75.10
start_time =  448.30
end_time =  505.78
start_time =  534.48
end_time =  595.46
start_time =  644.24
end_time =  697.58
start_time =  806.58
end_time =  869.58
start_time =  1119.58
end_time =  1169.58
start_time =  1253.58
end_time =  1298.58
start_time =  1899.52
end_time =  1974.78



1044
start_time =  0.00s -> 4.26s
end_time =  73.40s -> 75.10s
start_time =  448.30s -> 449.00s
end_time =  505.02s -> 505.78s
start_time =  534.48s -> 536.16s
end_time =  594.30s -> 595.46s
start_time =  644.24s -> 645.78s
end_time =  694.58s -> 697.58s
start_time =  806.58s -> 808.58s
end_time =  867.58s -> 869.58s
start_time =  1119.58s -> 1120.58s
end_time =  1166.58s -> 1169.58s
start_time =  1253.58s -> 1255.58s
end_time =  1295.58s -> 1298.58s
start_time =  1899.52s -> 1902.42s
end_time =  1972.64s -> 1974.78s

上記コード考える用　ここから

In [107]:
print(time[0])
time[0].split("s -> ")[0]

0.00s -> 4.26s


'0.00'

In [108]:
print(time[0])
time[0].split("s -> ")[1].split("s")[0]

0.00s -> 4.26s


'4.26'

In [109]:
for i in range(len(keyword_sentence_start)):
    print("From: ", keyword_sentence_start[i], " to: ", keyword_sentence_end[i])

From:  0.00  to:  75.10
From:  448.30  to:  505.78
From:  534.48  to:  595.46
From:  644.24  to:  697.58
From:  806.58  to:  869.58
From:  1119.58  to:  1169.58
From:  1253.58  to:  1298.58
From:  1899.52  to:  1974.78


上記コード考える用　ここまで

In [110]:
for i in range(len(keyword_sentence)):
    print(i, " from: ", keyword_sentence_start[i], " to: ", keyword_sentence_end[i]," \n", keyword_sentence[i])

0  from:  0.00  to:  75.10  
 ...これを張ってない会社っていうのはまずないはずなんだよね
やっぱり現場確認するっていうのはとても重要になってきます
ここは全部換気扇だと思ってください
家作りや人生作り
皆さんどうも職人社長の平松です
平松社長また現場に来てますね
そうですね
今回ですね上等からしばらく経って
断熱とか施工とかですね防水関係とかっていうところの
いろんな重要なポイントを解説プラス
実際手を動かして施工していきたいと思います
その手前の段階でシリーズになってるんで
その前の内容を知りたいというか見たいという方はですね
概要欄の方に貼っておくのでそちらのまた見ていくと
こういう感じで済んでるんだなって分かると思うので
またそちらもぜひ見ていただければと思います
ということで早速ですね今回は
断熱の施工の仕方とか通気の取り方とか
断熱性耐久性省エネ性能とかに大きく影響があるようなですね
細かい施工ちょっとの違いで大きく差が出てくるんですよ
っていうところが分かるようにね
解説をして手を動かしてやっていきますので
早速スタートしていきましょう
早速見ていきましょうの前に
12月28日に門川さんからスマイタイゼンという本が
出版されることになりました
おめでとうございます
ありがとうございます
皆さんがチャンネル登録したりと応援していただいているおかげでですね
...
1  from:  448.30  to:  505.78  
 ...こっちが両方
これが平らなところ
そこをテープでちゃんと防水もして
しっかりと入らないようにこうしてるというような感じです
基本的にさらに細かいことを言うと
シートは下から貼っていくんですね
当たり前なんです
この透湿防水シートっていうのは
家の下の方から貼っていく
水は高いところから上に行くんで
施工は逆に下から上が原則なんですよ
なぜかっていうと
下から貼るでしょ
上かぶせるでしょ
この分かぶせるんですね
この上から雨が落ちてきても
入りようがないってことですね
上から貼っていったら
全部この上にかぶせるのに全部入るじゃないですか
そうですね
一回だけそういう現場見たことある
社長がやってるんですか
自分はやってないから
えぇって思って
そういう現場見たことあるんで
稀な例だと思うけど
原理

In [111]:
len(keyword_sentence)

8